### ISSUE TITLE
[suggestion] parcool support

### ISSUE THREAD

**Issue by gitguette** [tr7zw/FirstPersonModel#381]
not sure how hard this would be, but adding support for changing the perspective in parcool animations would make it a lot more realistic. it feels weird to see my entire body while doing flips or dives.

**Comment by SamuelUlep** [comment]
alRex-U is attempting to fix the issue https://github.com/alRex-U/ParCool/issues/413 but requires curse maven third party distribution for an attempt to fix it, if tr7zw would allow it.

**Comment by tr7zw** [comment]
For the diving you need to offset the model to the camera, probably(there is an API for that https://github.com/tr7zw/FirstPersonModel/blob/main/src/main/java/dev/tr7zw/firstperson/api/PlayerOffsetHandler.java ). The shield is not Firstperson mod, thats NEA and rotation lock I assume. Firstperson mod does nothing to the player model itself, other than hiding the head and rendering it offset to the camera.

**Comment by tr7zw** [comment]
Neither Firstperson mod nor Not Enough Animations have dependencies. Firstperson is just designed to need NEA, but it works without. If its NEA, there might be a few options to add support.

**Comment by tr7zw** [comment]
Yea, it's only added as a dependency there, because running firstperson without NEA will cause a ton of visual issues (no maps, no rotation lock, bow not working well, no eating animation etc), so I have them as "logical" dependencies. But technically, they don't need each other. Since it's NEA, the best fix would be if you register an animation via https://github.com/tr7zw/NotEnoughAnimations/blob/main/src/main/java/dev/tr7zw/notenoughanimations/api/NotEnoughAnimationsApi.java that just claims ownership of all limbs and the body rotation while a parcool animation is playing. Then NEA shouldn't interfere with anything anymore.

**Comment by SamuelUlep** [comment]
I'll mention this two things to alRex-U about needing to claim limb ownership and the camera offset during the dive animation. It would probably be a nice thing to say in the mod page of FPM that NEA isn't really a hard dependency but also highly recommended to have a smoother experience (or something along those lines). Cheers!

**Comment by tr7zw** [comment]
Not sure if that's worth mentioning on the modpage, because why would you use it without NEA. Without NEA, Firstperson just doesn't work correctly (they used to be the same mod too). 
But yes, the API of both plugins should be enough to fix all issues with parcool.

**Comment by tr7zw** [comment]
Yep, both bad examples, as they don't conflict with NEA^^. NEA is fully configurable, so even if you use a pack that replaces everything and overlaps with NEA with stuff like the ladder animation/eating animation, you still need the mod to get things like map holding, even if you turn off everything else. Some stuff like the rotation lock/map holding can't be replicated by EMF packs.

**Comment by SamuelUlep** [comment]
I concede, I just thought it was something nice to have front and center.

**Comment by tr7zw** [comment]
I guess, if someone really wants to tinker with things, they will notice that it's not defined as a hard dependency 😅

**Comment by SamuelUlep** [comment]
They'll see this ig and see its nots a hard dependency for the mod
